query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -250,6 +256,30 @@ private String convertOperator2String(Operator operator) {
     }
   }
 
+  //called for selection columns
+  private String convertExpression2String(Expression expression, Params.Builder paramsBuilder,int selectionCounter) {
```

Comment:
This can't be a new requirement just for avg rate - I'm guessing we already have something like this or equivalent in here.

Reply:
Yes. This is already present. Just overloaded it with extra parameter.